### PR TITLE
12329 Pin mypy version somewhat

### LINKED ISSUE #12329: mypy suddenly started failing
Not sure why.

e.g. https://github.com/twisted/twisted/actions/runs/11276233177/job/31359524493

### PR DESCRIPTION
**Pull request by itamarst**
## Scope and purpose

Fixes #12329

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -128,8 +128,8 @@ gtk-platform = [
 
 mypy = [
     "twisted[dev,all-non-platform]",
-    "mypy~=1.8",
-    "mypy-zope~=1.0.3",
+    "mypy<1.11",
+    "mypy-zope",
```
Comment (adiroiban) [inline pyproject.toml]: My understanding is that the `twisted[mypy]` deps are only to be install for development work.

They don't need to be installed for "production"

So `twisted[mypy]` is kind of a private extra deps ...

This mean that we should be able to pin the exact version... nobody else should depend on `twisted[mypy]`

```suggestion
    "mypy==1.11",
    "mypy-zope==1.0.6",
```

**Review by adiroiban — APPROVED** [review]
Thanks for the fix

A left a minor comment.

I think that it's important to fix trunk ... and if needed, we can discuss deps pinning in https://github.com/twisted/twisted/issues/12114

File: pyproject.toml (hunk 1 above)
Comment (itamarst) [reply pyproject.toml]: Can't quite apply the suggestion because 1.11 is what broke it, so upgrading will be a separate issue.

**Comment by adiroiban** [comment]
Thanks again for the fix. Yep, my suggestion was off :)

**Pull request merged**